website/api/controllers/webhooks/receive-from-customer-fleet-instance.js

```diff
@@ -102,7 +106,7 @@ module.exports = {
       if(err.raw.statusCode === 404){
         return new Error(`When sending a request to unenroll a host from a Workspace One instance (Host information: Serial number: ${host.hardware_serial}, id: ${host.id}, uuid: ${host.uuid}), the specified host was not found on the customer's Workspace One instance. Full error: ${err.stack}`);
       } else if(err.raw.statusCode === 400) {
-        return new Error(`When sending a request to unenroll a host from a Workspace One instance (Host information: Serial number: ${host.hardware_serial}, id: ${host.id}, uuid: ${host.uuid}), the Workspace One instance could not unenroll the specified host. Full error: ${err.stack}`);
+        return { badRequest: `When sending a request to unenroll a host from a Workspace One instance (Host information: Serial number: ${host.hardware_serial}, id: ${host.id}, uuid: ${host.uuid}), the Workspace One instance could not unenroll the specified host. Full error: ${err.stack}` };
```

Comment:
Just curious, are these migrations webhooks in the fleetdm.com app limited to Workspace One?

Reply:
Yep, just WS1 for now from what I understand

Reply:
for this to work as expected, there needs  to be a `throw` rather than a `return`, i.e.:

`throw { badRequest: `… };`

Reply:
This isn't a `badRequest` from the requestor's perspective-- this is Workspace One saying to fleetdm.com that this was a bad request.   That means workspace one is telling us we're sending the wrong thing.  We need to find out why Workspace One is saying that, and then if this is a problem on their end, and they're just incorrectly sending a 4xx status code, then we can address it.  But otherwise, the 400 would indicate that Fleet's code is wrong and we are sending something that Workspace One doesn't know how to handle.

fyi @eashaw

Reply:
In the current flow this 
```mermaid
sequenceDiagram
  participant D as fleetd
  participant S as fleet server
  participant W as workspace1
  D->>S: POST webhook
  S->>W: POST to unenroll
  W->>S: 400 Error
  S->>D: This is the return we are discussing
  D->>S: /debug/errors (which currently throws a 500 on purpose for alerts)
```

What we want here is more of an indicator of an issue that fleetd can react on, These are possibly issues that maybe be with the configuration of the webhook but are not a failure to communicate. So 4xx is still a valid value for this webhook return. We also want to reduce the number of alerts that trigger p1 investigations since normally this response is not something infra can respond to.

I do see further down the switch case if it's not a 404 or 400 response from WS1 will still respond with a 500 error (which in that case we are the proxy to a backend gateway and it's correct)

If you agree with this let me know and we can move forward on changing this to a 400 response to the fleetd agent